pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -158,10 +160,44 @@ func (w *Watcher) invokeRegistrationCallbackAtHandler(ctx context.Context, clien
 	}); err != nil {
 		return fmt.Errorf("failed to send registration status at socket %s, err: %v", socketPath, err)
 	}
+
 	chanForAckOfNotification <- true
 	return nil
 }
 
+// Handle filesystem notify event.
+func (w *Watcher) handlefsnotifyEvent(event fsnotify.Event) error {
+	if event.Op&fsnotify.Create != fsnotify.Create {
+		return nil
+	}
+
+	fi, err := os.Stat(event.Name)
+	if err != nil {
+		return fmt.Errorf("stat file %s failed: %v", event.Name, err)
+	}
+
+	if !fi.IsDir() {
+		err := w.registerPlugin(event.Name)
```

Comment:
Hi @figo,

Please consider reducing the scope of `err` by combining it on the same line with the `if` statement.

Reply:
I'd argue doing so makes the code less readable, but I don't have a strong preference either way

Reply:
Hi @verult,

I just prefer to keep things less indented (less than 80 characters wide), but I'm fine with adhering to the project standard. It's certainly a personal preference.